resources/js/components/shared/TimelinePlot.vue

```diff
@@ -0,0 +1,450 @@
+      </span>
+    </div>
+  </div>
+</template>
+<script>
+import {ref, onMounted } from 'vue';
+export default {
+  name: "TimelinePlot",
+  /*
+   * This template generates a line plot for a group of time-series datasets,
+   * where each set is allowed to have a different number of data points.
+   *
+   * The expected input is of the form:
+   *   plotData: [
+   *   |   {
+   *   |   | color: <line color>,  <--- d3 accepts most color formats (hex, rgb, etc.)
+   *   |   | name: "<name to appear in legend>",
+   *   |   | values: [
+   *   |   |   {
+   *   |   |   | x: <Date object>,
+   *   |   |   | y: <numeric>,
+   *   |   |   | url: "<url>"      <--- optional, renders clickable points if provided
+   *   |   |   },
+   *   |   |   ...
+   *   |   | ],
+   *   |   },
+   *   |   ...
+   *   | ],
+   *   },
+   *   title: "<plot title>",
+   *   xLabel: "<x-axis label>",
+   *   yLabel: "<y-axis label>"
+   */
+
+  props: {
+    plotData: {
+      type: Object,
+      required: true,
+      default: () => { return null }
+    },
+    title: {
+      type: String,
+      required: true,
+      default() {
+        return "Timeline Plot"
+      },
+    },
+    xLabel: {
+      type: String,
+      required: true,
+      default() {
+        return "Time"
+      },
+    },
+    yLabel: {
+      type: String,
+      required: true,
+      default() {
+        return "Value"
+      },
+    },
+  },
+
+  setup () {
+    // initialize refs
+    const divRef = ref(); // div element in which to render the plot
+    const tooltipRow1Axis = ref();
+    const tooltipRow1Value = ref();
+    const tooltipRow2Axis = ref();
+    const tooltipRow2Value = ref();
+    onMounted(() => {
+      divRef.value;
+      tooltipRow1Axis.value;
+      tooltipRow1Value.value;
+      tooltipRow2Axis.value;
+      tooltipRow2Value.value;
+    });
+  },
+
+  mounted () {
+      this.plot_timeline_graph();
+  },
+
+  methods: {
+    plot_timeline_graph() {
+
+      const plot_div = this.$refs.divRef;
+      const data = this.plotData.map((d) => { return d['values'] });
+      const colors = this.plotData.map((d) => { return d.color });
+      const line_names = this.plotData.map((d) => { return d.name });
+      const title = this.title;
+      const xLabel = this.xLabel;
+      const yLabel = this.yLabel;
+      const tooltipRow1Axis = this.$refs.tooltipRow1Axis;
+      const tooltipRow1Value = this.$refs.tooltipRow1Value;
+      const tooltipRow2Axis = this.$refs.tooltipRow2Axis;
+      const tooltipRow2Value = this.$refs.tooltipRow2Value;
+
+
+      // ################## INPUT DATA (temporary) ##################
+      // TODO: refactor all `.style()` calls to CSS sheet
+      const legend_font_size = 18; // GIVEN
+      const title_font_size = 22; // GIVEN
+
+      // TODO: fit relative to page?
+      // define plot size
+      const margin = {top: 2*title_font_size, right: 30, bottom: 30, left: 60};
```

Comment:
@williamjallen  Good point. Interestingly enough, this one wasn't among the long list of things eslint was complaining about it my code. We could consider changing the settings so that this is enforced.

Reply:
Good point.  Now that we have eslint, things like this should be automated rather than relying upon different people with different opinions catching it in code review.

The [`object-property-newline`](https://eslint.org/docs/latest/rules/object-property-newline) rule appears to be the one we want.  I'll make a PR to add it.